shared/rds_instance/main.tf

```diff
@@ -0,0 +1,92 @@
+# Required variables:
+variable "name" {
+  description = "The name for this database (usually the application name)."
+}
+
+variable "instance_class" {
+  description = "The RDS instance class. See: https://goo.gl/vTMqx9"
+}
+
+# Optional variables:
+variable "allocated_storage" {
+  description = "The amount of storage to allocate to the database, in GB."
+  default     = 100
+}
+
+variable "subnet_group" {
+  description = "The AWS subnet group name for this database."
+  default     = "default-vpc-7899331d"
+}
+
+variable "security_groups" {
+  description = "The security group IDs for this database."
+  default     = ["sg-c9a37db2"]
+}
+
+# Data resources:
+data "aws_ssm_parameter" "database_username" {
+  name = "/${var.name}/rds/username"
+}
+
+data "aws_ssm_parameter" "database_password" {
+  name = "/${var.name}/rds/password"
+}
+
+resource "aws_db_instance" "database" {
+  identifier = "${var.name}"
+  name       = "longshot"
+
+  engine            = "mariadb"
+  engine_version    = "10.3"
+  instance_class    = "${var.instance_class}"
+  allocated_storage = "${var.allocated_storage}"
+
+  allow_major_version_upgrade = true
+
+  backup_retention_period = 7             # 7 days.
+  backup_window           = "06:00-07:00" # 1-2am ET.
+
+  username = "${data.aws_ssm_parameter.database_username.value}"
+  password = "${data.aws_ssm_parameter.database_password.value}"
+
+  # TODO: We should migrate our account out of EC2-Classic, create
+  # a default VPC, and let resources be created in there by default!
+  db_subnet_group_name = "${var.subnet_group}"
+
+  vpc_security_group_ids = "${var.security_groups}"
+  publicly_accessible    = true
+
```

Comment:
Did we want to add `deletion_protection  = true` as a default value, or keep as `false`, and specify `true` manually for Prod instances?

Reply:
Ah yes!! Let me add that in a follow-up PR so this one stays as a no-op.